src/plan/gencopy/mutator.rs

```diff
@@ -0,0 +1,70 @@
+use super::gc_works::*;
+use super::GenCopy;
+use crate::plan::barriers::*;
+use crate::plan::mutator_context::Mutator;
+use crate::plan::mutator_context::MutatorConfig;
+use crate::plan::Allocator as AllocationType;
+use crate::util::alloc::allocators::{AllocatorSelector, Allocators};
+use crate::util::alloc::BumpAllocator;
+use crate::util::OpaquePointer;
+use crate::vm::VMBinding;
+use crate::MMTK;
+use enum_map::enum_map;
+use enum_map::EnumMap;
+
+pub fn gencopy_mutator_prepare<VM: VMBinding>(
+    _mutator: &mut Mutator<GenCopy<VM>>,
+    _tls: OpaquePointer,
+) {
+    // Do nothing
+}
+
+pub fn gencopy_mutator_release<VM: VMBinding>(
+    mutator: &mut Mutator<GenCopy<VM>>,
+    _tls: OpaquePointer,
+) {
+    // rebind the allocation bump pointer to the nursery space
+    let bump_allocator = unsafe {
+        mutator
+            .allocators
+            .get_allocator_mut(mutator.config.allocator_mapping[AllocationType::Default])
+    }
+    .downcast_mut::<BumpAllocator<VM>>()
+    .unwrap();
+    bump_allocator.rebind(Some(&mutator.plan.nursery));
+}
+
+lazy_static! {
+    pub static ref ALLOCATOR_MAPPING: EnumMap<AllocationType, AllocatorSelector> = enum_map! {
+        AllocationType::Default => AllocatorSelector::BumpPointer(0),
+        AllocationType::Immortal | AllocationType::Code | AllocationType::ReadOnly => AllocatorSelector::BumpPointer(1),
```

Comment:
`gencopy` is only for openjdk and cannot run on jikesrvm (no barriers support). It is correct for openjdk since openjdk has no immortal/code spaces. I'll leave a TODO comment here about this.

Reply:
Done